python/lzreposync/src/lzreposync/primary_parser.py

```diff
@@ -0,0 +1,220 @@
+            raise ValueError("No package being parsed")
+
+        elt_name = node.localName
+        dependencies: [Dependency] = []
+        for child_node in node.childNodes:
+            if child_node.nodeType == node.ELEMENT_NODE:
+                dependency = Dependency()
+                for attr_name in dependency.attributeTypes.keys():  # ['name', 'version', 'flags']
+                    if child_node.attributes and attr_name not in child_node.attributes.keys():
+                        continue  # Skipping Eg: 'epoch'
+                    dependency[attr_name] = child_node.getAttributeNode(
+                        "ver" if attr_name == "version" else attr_name  # mapping attr name
+                    ).value
+                dependencies.append(dependency)
+        self.currentPackage[elt_name] = dependencies
+
+    def set_text_element_node(self, node):
+        """
+        Parse and set elements with text content. Eg: <summary>GStreamer ...</summary>
+        """
+        if not self.currentPackage:
+            print("Error: No package being parsed!")
+            raise ValueError("No package being parsed")
+
+        self.currentPackage[node.localName] = get_text(node)
+
+    def set_element_node(self, node):
+        """
+        Parse the given node and the corresponding information to the corresponding package's attribute
+        """
+        if not self.currentPackage:
+            print("Error: No package being parsed!")
+            raise ValueError("No package being parsed")
+
+        if node.nodeType == node.ELEMENT_NODE and node.namespaceURI == COMMON_NS and node.localName == "format":
+            # Recursively set the child elements of the <format> element
+            for child in node.childNodes:
+                self.set_element_node(child)
+        if node.nodeType == node.ELEMENT_NODE and node.namespaceURI == COMMON_NS and node.localName == "checksum":
+            self.set_checksum_node(node)
+        if node.nodeType == node.ELEMENT_NODE and node.hasAttributes():
+            # node in searchedAttrs
+            self.set_attribute_element_node(node)
+        elif node.nodeType == node.ELEMENT_NODE and not is_complex(node.localName):
+            # node in searchedChars:
+            self.set_text_element_node(node)
+        elif node.nodeType == node.ELEMENT_NODE and is_complex(node.localName):
+            # dealing with ["provides", "requires", "enhances", "obsoletes"]
+            self.set_complex_element_node(node)
+
+    def set_pacakge_header(self, node):
+        """
+        Set the package header. Mainly 'is_source' and 'packaging'
+        """
+        if not self.currentPackage:
+            print("Error: No package being parsed!")
+            raise ValueError("No package being parsed")
+
+        self.currentPackage['header'] = RPMHeader()
+        arch_node = node.getElementsByTagName("arch")[0]
+        if get_text(arch_node) == "src":
+            self.currentPackage['header'].is_source = True
```

Comment:
Be ware that there is another class for source packages in importlib: `SourcePackage`. Just setting the header may not be enough